src/librustc_mir/borrow_check/move_errors.rs

```diff
@@ -293,7 +293,7 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
     fn add_move_hints(
         &self,
         error: GroupedMoveError<'tcx>,
-        err: &mut DiagnosticBuilder<'a>,
+        err: &mut DiagnosticBuilder<'_>,
```

Comment:
This looks like an actual change since `'a` was introduced in `impl MirBorrowckCtxt<'a, 'gcx, 'tcx> {` above, no?

Reply:
Probably, yes -- however, arguably, it's a "correct" change as this branch compiles. If we don't need to constrain the lifetime, it's easier to understand the code if we're not doing so.

Reply:
And yet, if it compiles, seems fine.